sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -80,11 +80,7 @@ class RelationalGroupedDataset protected[sql](
     }
   }
 
-  // Wrap UnresolvedAttribute with UnresolvedAlias, as when we resolve UnresolvedAttribute, we
-  // will remove intermediate Alias for ExtractValue chain, and we need to alias it again to
-  // make it a NamedExpression.
```

Comment:
The comment is wrong as we don't remove top-level aliases for aggregate expressions. It causes problems as it wraps `UnresolvedAttribute` with `UnresolvedAlias`, making it not top-level anymore. Then the alias will be removed after this patch and `UnresolvedAlias` generates a different name.

For nested field `a.b`, previously the resolved expression is `Alias(GetStructField(...), "b")` and the `Alias` is not removed. `UnresolvedAlias` is useless and will be simply removed. So the final output column name is `b`. Now we remove the `Alias`, and `UnresolvedAlias` kicks in and generates a new `Alias` with the name `a.b`, which is a behavior change.

Here I simply remove this `UnresolvedAlias`, to make the behavior the same as before.

Reply:
Hmm, this code is pretty old, 2015.

I saw `alias` is also used to add alias around grouping expressions, not just aggregate expressions. Seems the comment is more for the case?

Reply:
This is only for `Aggregate.aggregateExpressions` not `Aggregate.groupingExpressions`. `Aggregate.aggregateExpressions` can include grouping expressions, but it doesn't matter. It needs to be `Seq[NamedExpression]` and Spark won't remove the top-level alias in it.